src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -361,7 +361,7 @@ trait ImplicitRunInfo { self: RunInfo =>
         computeIScope(cacheResult = false)
       else implicitScopeCache get tp match {
         case Some(is) => is
-        case None => computeIScope(cacheResult = true)
+        case None => computeIScope(cacheResult = seen.isEmpty)
```

Comment:
This greatly reduces the effectiveness of the cache since `seen` is almost never empty. I'm OK with merging this now because I have an old WIP branch where I try to fix this and other implicit issues that I will rebase and finish once I'm done with the incremental compilation stuff: https://github.com/dotty-staging/dotty/commits/fix/implicit-caching